src/crimson/osd/pg_backend.cc

```diff
@@ -830,6 +830,107 @@ PGBackend::get_attr_ierrorator::future<> PGBackend::get_xattrs(
     return get_attr_errorator::now();
   });
 }
+static inline int do_xattr_cmp_str(int op, string& v1s, bufferlist& xattr)
+{
+  string v2s(xattr.c_str(), xattr.length());
+
+  logger().debug("do_xattr_cmp_str '{}' vs '{}' op {}", v1s, v2s, op);
+
+  switch (op) {
+  case CEPH_OSD_CMPXATTR_OP_EQ:
+    return (v1s.compare(v2s) == 0);
+  case CEPH_OSD_CMPXATTR_OP_NE:
+    return (v1s.compare(v2s) != 0);
+  case CEPH_OSD_CMPXATTR_OP_GT:
+    return (v1s.compare(v2s) > 0);
+  case CEPH_OSD_CMPXATTR_OP_GTE:
+    return (v1s.compare(v2s) >= 0);
+  case CEPH_OSD_CMPXATTR_OP_LT:
+    return (v1s.compare(v2s) < 0);
+  case CEPH_OSD_CMPXATTR_OP_LTE:
+    return (v1s.compare(v2s) <= 0);
+  default:
+    return -EINVAL;
+  }
+}
+
+static inline int do_xattr_cmp_u64(int op, __u64 v1, bufferlist& xattr)
```

Comment:
```suggestion
static int do_xattr_cmp_u64(int op, uint64_t lhs, bufferlist& rhs_xattr)
```

instead of using using linux types, i'd suggest use types defined by the C++ standard.

also, instead of using `v1` and `v2`, better off using variable names with some meaning. in this case, `lhs` and `rhs` stand for left-hand side and right-hand side. IMHO, they more well known shorthands used in mathematics.

and `inline` is but a hint, i think we should leave it to compiler to decide to inline the function or not.